yew-macro/src/html_tree/html_component.rs

```diff
@@ -406,6 +406,13 @@ impl Parse for Props {
                 return Err(syn::Error::new_spanned(&prop.label, "expected identifier"));
             }
 
+            if prop.label.optional.is_some() {
+                return Err(syn::Error::new_spanned(
+                    &prop.label,
+                    "optional properties are only supported on HTML elements. Yew components can use `Option<T>` properties to accomplish the same thing.",
```

Comment:
Is this convention mentioned anywhere? The [docs speak of elements](https://yew.rs/docs/concepts/html/elements).

Reply:
From the original RFC https://tools.ietf.org/html/rfc1866#page-6: 
> tag
> Markup that delimits an element. A tag includes a name
> which refers to an element declaration in the DTD, and
> may include attributes. [SGML]

> element
> A component of the hierarchical structure defined by a
> document type definition; it is identified in a document
> instance by descriptive markup, usually a start-tag and
> end-tag. [SGML]

Reply:
I guess this boils down to how you look at it.
On the syntax level `<div />` is referred to as a tag in Yew but the object it's representing is an html element.
Does that make any sense?

Most other macro error messages use "tag" but the naming convention in Yew isn't really that strict.
For instance, we have this confusing error message which uses "element" instead of "attribute":
> Only a single key element is allowed on a <></>

Feel free to go with whichever you feel is clearer.